src/status_im2/contexts/wallet/home/view.cljs

```diff
@@ -24,25 +25,52 @@
       :sub-label           (i18n/label :t/add-address-description)
       :add-divider?        true}]]])
 
-(def account-cards
-  [{:name                "Account 1"
-    :balance             "€0.00"
-    :percentage-value    "€0.00"
-    :customization-color :blue
-    :type                :empty
-    :emoji               "🍑"
-    :on-press            #(rf/dispatch [:navigate-to :wallet-accounts])}
-   {:customization-color :blue
-    :on-press            #(rf/dispatch
-                           [:show-bottom-sheet {:content new-account}])
-    :type                :add-account}])
+(def add-account-placeholder
+  {:customization-color :blue
+   :on-press            #(rf/dispatch [:show-bottom-sheet {:content new-account}])
+   :type                :add-account})
 
 (def tabs-data
   [{:id :assets :label (i18n/label :t/assets) :accessibility-label :assets-tab}
    {:id :collectibles :label (i18n/label :t/collectibles) :accessibility-label :collectibles-tab}
    {:id :activity :label (i18n/label :t/activity) :accessibility-label :activity-tab}])
 
-(defn view
+(defn calculate-raw-balance
+  [rawBalance decimals]
+  (/ (js/parseInt rawBalance) (Math/pow 10 (js/parseInt decimals))))
+
+(defn calculate-balance
+  [{:keys [address]}]
+  (let [tokens       (rf/sub [:wallet-2/tokens])
+        token        (get tokens (keyword (string/lower-case address)))
+        total-values (atom 0)]
+    (doseq [item token]
```

Comment:
this is more like js style, this should be done in more idiomatic way

Reply:
GPT-3.5

In Clojure, it's generally preferred to avoid using mutable state like atoms and favor a functional style whenever possible. Here's a more idiomatic Clojure version of your code using reduce and other functional constructs:

clojure
Copy code
(defn calculate-total-value [item]
  (let [total-value-per-token
        (reduce
          (fn [acc balances]
            (let [raw-balance (:rawBalance balances)
                  decimals (:decimals item)]
              (+ acc (* (calculate-raw-balance raw-balance decimals)
                         (get-in item [:marketValuesPerCurrency :USD :price])))))
          0
          (vals (:balancesPerChain item)))]
    (+ total-value-per-token @total-values)))

(doseq [item token]
  (swap! total-values + (calculate-total-value item)))
In this version:

We define a calculate-total-value function that takes an item as an argument and calculates the total value for that item using reduce. This avoids the use of mutable atoms within the inner loop.

The reduce function accumulates the total value by iterating over the balances of the item, calculating the raw balance, and summing the values based on the item's market value per USD.

We use swap! to update the total-values atom by adding the calculated total value for each item in the doseq loop.

This version is more idiomatic and functional, avoiding mutable state and favoring data transformation operations like reduce.

Reply:
it's not bad, but second doseq also should be avoided

Reply:
I removed all the doseq and the atoms and used `reduce` function

Reply:
Done